crates/cast/tests/utils/build_contracts.sh

```diff
@@ -1,31 +1,19 @@
 #!/bin/bash
 set -e
 
-# "major-cairo-version scarb-version"
-VERSIONS=("v1 0.4.1" "v2 0.5.2")
-ASDF_DATA_DIR=$(asdf info | grep -e "ASDF_DATA_DIR" | awk -F '=' '{print $2}')
+SCARB_VERSION="0.5.2"
+CONTRACTS_DIRECTORY="$(git rev-parse --show-toplevel)/crates/cast/tests/data/contracts/"
 
-for versions_tuple in "${VERSIONS[@]}"; do
-  read -r cairo_version scarb_version <<< "$versions_tuple"
+if ! scarb --version | grep -qF "$SCARB_VERSION"; then
+  echo "Please run tests/utils/prepare_for_tests.sh script first!"
+  echo "wrong version of scarb found, required version is $SCARB_VERSION"
+  exit 1
+fi
 
-  CONTRACTS_DIRECTORY="$(git rev-parse --show-toplevel)/crates/cast/tests/data/contracts/$cairo_version"
-  SCARB_BIN="$ASDF_DATA_DIR/installs/scarb/$scarb_version/bin/scarb"
-
-  pushd "$CONTRACTS_DIRECTORY"
-  asdf local scarb "$scarb_version"
-  popd
-
-  if command -v "$SCARB_BIN" &> /dev/null; then
-    for contract_dir in "$CONTRACTS_DIRECTORY"/*; do
-      if ! test -d "$contract_dir"/target && [[ "$contract_dir" != *"fails"* ]]; then
-        pushd "$contract_dir"
-        scarb build
-        popd
-      fi
-    done
-
-  else
-    echo "Please run tests/utils/prepare_for_tests.sh script first!"
+for contract_dir in "$CONTRACTS_DIRECTORY"/*; do
+  if ! test -d "$contract_dir"/target && [[ "$contract_dir" != *"fails"* ]]; then
```

Comment:
Now I wonder if we should exclude already built directories. If someone is running this script he usually want to build them

Reply:
This script is being called by our test suite if I remember correctly, not sure compiling these contracts each time we want to run test is preferable (especially so that these contracts generally don't change)